### PR TITLE
[NO QA] Call Web-E API to get temporary token in Mapbox setup script

### PR DESCRIPTION
**Pull request by hayata-suenaga**
@neil-marcellini @tgolen 
cc: @fabioh8010 @blazejkustra because this PR involves TypeScript code

### Details

Most of the context behind this PR is covered in the issue description [here](https://github.com/Expensify/Expensify/issues/309451).

This PR gets rid of existing bash scripts that I added in [this PR](https://github.com/Expensify/App/pull/25506). These bash scripts helped contributors

### Fixed Issues

$ https://github.com/Expensify/Expensify/issues/309451 (for calling a Web-E endpoint)
$ https://github.com/Expensify/App/issues/25442 (for re-writing bash scripts in node/typescript)

### Tests

For engineers: 
1. Check out the branch `hayata-change-mapbox-setup-script`
2. Change [this line](https://github.com/Expensify/App/blob/0c7eef32d4e078c44ac956b72f7374d3e82b1e43/scripts/setup-mapbox-sdk.ts#L5) as below:
    ```ts
    const TOKEN_ENDPOINT = 'https://my-json-server.typicode.com/hayata-suenaga/mapbox-mock-response/token';
    ```
    _You can skip this step if [this issue is closed](https://github.com/Expensify/Expensify/issues/309417) and both [this PR](https://github.com/Expensify/Salt/pull/12649) and [this PR](https://github.com/Expensify/Web-Expensify/pull/38559) are merged. Otherwise, this step ensures that the script calls a mock API that I set up temporarily to simulate the response that will be returned once new API command in Web-E is deployed._

3. Run `npm run configure-mapbox`
4. Open `~/.netrc` and confirm that the file includes the content below:
    ```txt
    machine api.mapbox.com
    login mapbox
    password example-token
    ```
5. Open `~/.gradle/gradle.properties` and confirm that the file includes the content below:
    ```txt
    MAPBOX_DOWNLOADS_TOKEN=example-token
    ```
6. Repeat steps 3 ~ 5 and confirm that there isn't a duplicate entry in `.netrc` and `gradle.properties`

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] If we are not using the full Onyx data that we loaded, I've added the proper selector in order to ensure the component only re-renders when the data it is using changes
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

</details>

<details>
<summary>Mobile Web - Chrome</summary>

</details>

<details>
<summary>Mobile Web - Safari</summary>

</details>

<details>
<summary>Desktop</summary>

</details>

<details>
<summary>iOS</summary>

</details>

<details>
<summary>Android</summary>

</details>

### CODE REVIEW

**Review by allroundexperts — COMMENTED** [review]
Great job @hayata-suenaga 🎖️ 🎖️ 🎖️

Requested some minor changes.

File: package.json
Comment (hayata-suenaga) [reply package.json:211]: ah I didn't know `fetch` exits in node run time. trying to see if fetch works

Comment (hayata-suenaga) [reply package.json:211]:
```
hayatasuenaga@Hayatas-Work-MBP App % npm run configure-mapbox

> new.expensify@1.3.55-7 configure-mapbox
> ts-node scripts/setup-mapbox-sdk-local.ts

Error: fetch is not defined
```

Fetch API was apparently introduced in Node 17. App repo somehow still uses node 16 (see the `engines` field in `package.json`) 😭 so `fetch` is not available.

_ts-node uses the node that is active in the current terminal session, which is node 16 in the App repo_

Comment (hayata-suenaga) [reply package.json:211]: The other option is to use `https` module from node but the syntax is kind of verbose (even considering that we only need to make one API call). So, I'm thinking of sticking with `axios`. What do you think @allroundexperts

File: scripts/setup-mapbox-sdk.ts
Comment (hayata-suenaga) [reply scripts/setup-mapbox-sdk.ts]: agree! changed

File: scripts/setup-mapbox-sdk.ts
Comment (hayata-suenaga) [reply scripts/setup-mapbox-sdk.ts]: I'm actually not a huge fun of typing API responses. I like leaving response values as `any` or `undefined` to do type narrowing.

I made a change in this commit to add type checking for type narrowing: https://github.com/Expensify/App/pull/25545/commits/5497523c9f8315a807bd8e0b4563414608fa1a88

File: .eslintignore
Comment (hayata-suenaga) [reply .eslintignore:4]: I think because these are node scripts and have different requirements/environment so existing ESLint rules might not make sense so I decided to exclude them from linting for now

**Comment by hayata-suenaga** [comment]
Test build is in process: https://github.com/Expensify/App/actions/runs/5922385154

File: scripts/setup-mapbox-sdk-ci.ts
Comment (fabioh8010) [inline scripts/setup-mapbox-sdk-ci.ts:17]:
```suggestion
if (token) {
```

I think we can simplify here.

File: scripts/setup-mapbox-sdk-local.ts
Comment (fabioh8010) [inline scripts/setup-mapbox-sdk-local.ts:27]:
```suggestion
            console.error('Error: ', error.message);
```

File: scripts/setup-mapbox-sdk.ts
Comment (fabioh8010) [inline scripts/setup-mapbox-sdk.ts:73]:
```suggestion
            console.error('Error: ', error.message);
```

File: scripts/setup-mapbox-sdk-local.ts
Comment (hayata-suenaga) [reply scripts/setup-mapbox-sdk-local.ts:27]: All console family (log, error, etc) adds a while space between arguments.

Screenshot 2023-08-21 at 7 21 26 AM

File: scripts/setup-mapbox-sdk-ci.ts
Comment (hayata-suenaga) [reply scripts/setup-mapbox-sdk-ci.ts]: nice suggestion changed!

**Review by blazejkustra — APPROVED** [review]
Looking good in terms of Typescript! 💯

**Comment by hayata-suenaga** [comment]
For this build failure, I apparently selected a wrong branch. I'm re-running the test build with the correct branch [here](https://github.com/Expensify/App/actions/runs/5927603323)

**Comment by hayata-suenaga** [comment]
okay the test builds succeeded 🚀

File: scripts/setup-mapbox-sdk-local.ts
Comment (fabioh8010) [reply scripts/setup-mapbox-sdk-local.ts:27]: You are right @hayata-suenaga !

File: scripts/setup-mapbox-sdk.ts
Comment (fabioh8010) [reply scripts/setup-mapbox-sdk.ts:73]: Thanks!

**Review by fabioh8010 — APPROVED** [review]
LGTM!

**Comment by hayata-suenaga** [comment]
### NOTE for myself

Fix this: https://expensify.slack.com/archives/C03TQ48KC/p1692745804074509

File: .eslintignore
Comment (tgolen) [reply .eslintignore:4]: I don't think we should ignore them. Node scripts are just like any other JS script and the same formatting and styles should apply to them. We've seen this in the GitHub actions as well. It makes sense that JS code everywhere is consistent, regardless of the platform it's running on.

File: package.json
Comment (tgolen) [reply package.json:211]: Since there is only one API call, I think using `https` is fine. Yes, it's more verbose, but adding this package just to make one API call is overkill.

File: scripts/setup-mapbox-sdk.ts
Comment (tgolen) [inline scripts/setup-mapbox-sdk.ts:40]: Don't use `async` and `await`. Those aren't approved patterns yet. This should use promises.

File: scripts/setup-mapbox-sdk.ts
Comment (tgolen) [inline scripts/setup-mapbox-sdk.ts:54]: If you're using `chmodSync()` you don't need `await` anyways, right? It's already synchronous.

**Review by tgolen — CHANGES_REQUESTED** [review]

**Comment by hayata-suenaga** [comment]
By the way, this is expected because [the new Web-E API command](https://github.com/Expensify/Web-Expensify/pull/38559) hasn't been deployed yet.

**Comment by tgolen** [comment]
@hayata-suenaga Is this ready for review again? It looks like there are a couple of conflicts.

**Comment by grgia** [comment]
bump on merge conflicts

**Comment by hayata-suenaga** [comment]
thank you for reminding me @grgia we're currently treating this as lower priority so I gonna go back to this PR later this week 🙇

**Comment by hayata-suenaga** [comment]
Changing the priority to LOW because this is not a necessity right now.

This mostly improves the setup that new contributors need to do.

Because this PR will affect the build process, we want to be sure before merge

**Comment by tgolen** [comment]
@hayata-suenaga What is the plan for this? Should the PR just be closed?

**Comment by hayata-suenaga** [comment]
I was originally planning to resume working on and merge this after Money2020

I don't know how much we need this PR. I think it's still nice to have feature. What do you think @tgolen ?

**Comment by tgolen** [comment]
I think I would just close this. If anyone feels it is painful enough to solve, we can reopen at a later time.

**Comment by hayata-suenaga** [comment]
according to [this comment](https://github.com/Expensify/App/pull/25545#issuecomment-1785228852), I'm closing this for now.

**Comment by hayata-suenaga** [comment]
sure I'm sorry that I didn't realize that. Asked for the payment to you [here](https://github.com/Expensify/App/issues/25442#issuecomment-1787676095)

**Pull request closed without merging**
